File: hyacinth/src/main/java/com/github/saturn_xiv/palm/hyacinth/models/Log.java
package com.github.saturn_xiv.palm.hyacinth.models;

import java.io.Serializable;
import java.util.Date;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;

import com.fasterxml.jackson.annotation.JsonProperty;

@Entity
@Table(name = "logs")
public class Log implements Serializable {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;
    @Column(nullable = false, name = "package")
    @JsonProperty("package")
    private String package_;
    @Column(nullable = false)
    private String service;
    @Column(nullable = false)
    private int timeTaking;
    @Column(nullable = false)
    private Date createdAt;

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getPackage_() {
        return package_;
    }

    public void setPackage_(String package_) {
        this.package_ = package_;
    }

    public String getService() {
        return service;
    }

    public void setService(String service) {
        this.service = service;
    }

    public int getTimeTaking() {
        return timeTaking;
    }

    public void setTimeTaking(int timeTaking) {
        this.timeTaking = timeTaking;
    }

    public Date getCreatedAt() {
        return createdAt;
    }

    public void setCreatedAt(Date createdAt) {
        this.createdAt = createdAt;
    }

}

File: dahlia/src/main/kotlin/com/github/saturn_siv/palm/dahlia/plugins/bookkeeper/models/Log.kt
package com.github.saturn_siv.palm.dahlia.plugins.bookkeeper.models

import com.github.saturn_siv.palm.dahlia.plugins.cms.models.User
import jakarta.persistence.*
import java.time.Instant

@Table(name = "bookkeeper_logs")
@Entity(name = "bookkeeper.log")
class Log(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(nullable = false)
    var id: Int,
    @Column(nullable = false)
    var operator: String,
    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    var type: Type,
    @Column(nullable = false)
    var ip: String,
    @Column(nullable = false)
    var message: String,
    @Column(nullable = false)
    var createdAt: Instant,
    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "team_id")
    var team: Team,
    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "book_id")
    var book: Book,
    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "user_id")
    val user: User,
) {
    enum class Type {

    }
}